fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -1253,6 +1254,20 @@ private static void checkSearchParameterCombinations(Class<?> resourceType, List
                     parameterValue.setValueSystem(unescapeSearchParm(parts[0]));
                     parameterValue.setValueCode(unescapeSearchParm(parts[1]));
                 } else {
+                    // Optimization for search parameters that always reference the same system, added under #1929
+                    try {
+                        String implicitSystem = searchParameter.getExtension().stream()
+                                .filter(e -> SearchConstants.IMPLICIT_SYSTEM_EXT_URL.equals(e.getUrl()) && e.getValue() != null)
+                                .findFirst()
+                                .map(e -> e.getValue().as(Uri.class).getValue())
+                                .orElse(null);
+                        if (implicitSystem != null) {
+                            parameterValue.setValueSystem(implicitSystem);
+                        }
```

Comment:
this is the meat of the change.  it just takes advantage of the extension that we've added to the SearchParameter definitions (when possible)